File: 08_SpeedOfBuilds/ConcretePractice/SlowTestsCodeFiles/CSharp/Starter/SpeedOfBuilds/Customer.cs
using System;
using System.Collections.Generic;
using System.Threading;

namespace SpeedOfBuilds
{
    public class Customer
    {


        private List<Order> orders = new List<Order>();




        public Customer(List<Order> orders)
        {

            this.orders = orders;

        }

        public Order FindBiggestOrder()
        {

            Order biggestOrder = orders[0];

            foreach (Order nextOrder in orders)
            {
                if (CompareOrder(nextOrder, biggestOrder) > 0)
                {
                    biggestOrder = nextOrder;

                    try
                    {
                        Thread.Sleep(1);
                    }
                    catch (Exception e)
                    {
                        Console.WriteLine(e);
                    }
                }
            }

            return biggestOrder;
        }




        public Order FindOrderByIndex(int index)
        {

            return orders[index];
        }


        public int CompareOrder(Order first, Order other)
        {
            if (first.Quantity > other.Quantity)
            {

                return 1;
            }
            else if (first.Quantity > other.Quantity)
            {
                return -1;
            }
            else
            {
                return 0;
            }

        }





    }
}

File: 08_SpeedOfBuilds/ConcretePractice/SlowTestsCodeFiles/Java/Demo/SpeedOfBuilds/src/main/java/com/qa/Customer.java
package com.qa;

import java.util.ArrayList;
import java.util.List;

public class Customer {
	
	
	private List<Order> orders = new ArrayList();
	
	
	
	
	public Customer(List<Order> orders) {
		
		this.orders = orders;
		
	}
	
	public Order findBiggestOrder() {
		
		Order biggestOrder = orders.get(0);
		
		for(Order nextOrder : orders)
		{
			if(compareOrder(nextOrder, biggestOrder) > 0) {
				biggestOrder = nextOrder;
				
				try {
					Thread.sleep(1);
				}
				catch(InterruptedException e) {
					System.out.println(e);
				}
			}
		}
		
		return biggestOrder;
	}
	
	
	
	
      public Order findOrderByIndex(int index) {
		
		return orders.get(index);
	 }
	
	
	public int compareOrder(Order first, Order other)
	{
		if(first.getQuantity()> other.getQuantity()) {
			
			return 1;
		}
		else if(first.getQuantity()> other.getQuantity()) {
			return -1;
		}
		else
		{
			return 0;
		}
	
	}
	
	
	
	

}
